File: frontend/src/types/annotation.ts
export type AnnotationType = 'polygon' | 'rectangle' | 'ellipse' | 'point'

export interface AnnotationLabel {
  id: string
  name: string
  color: string
}

export interface Annotation {
  id: string
  label_id: string
  type: AnnotationType
  points?: number[] // [x1, y1, x2, y2, ...]
  bbox?: [number, number, number, number] // [x, y, w, h]
  center?: [number, number] // [cx, cy]
  radius?: [number, number] // [rx, ry]
}

export interface AnnotationData {
  labels: AnnotationLabel[]
  annotations: Annotation[]
}

export type EditorMode = 'select' | 'hand' | 'polygon' | 'rectangle' | 'ellipse' | 'point' | 'eraser'

File: backend/app/schemas/annotation.py
from typing import List, Optional, Tuple, Union
from uuid import UUID
from pydantic import BaseModel, Field
from enum import Enum

class AnnotationType(str, Enum):
    POLYGON = "polygon"
    RECTANGLE = "rectangle"
    ELLIPSE = "ellipse"
    POINT = "point"

class AnnotationLabel(BaseModel):
    id: str
    name: str
    color: str

class Annotation(BaseModel):
    id: str
    label_id: str
    type: AnnotationType
    points: Optional[List[float]] = None
    bbox: Optional[Tuple[float, float, float, float]] = None
    center: Optional[Tuple[float, float]] = None
    radius: Optional[Tuple[float, float]] = None

class AnnotationData(BaseModel):
    labels: List[AnnotationLabel]
    annotations: List[Annotation]
